### ISSUE TITLE
🐛 [BUG]: The content option in your Tailwind CSS configuration is missing or empty

### ISSUE THREAD

**Issue by jesusvallez** [nxtensions/nxtensions#447]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What package does the issue affects?

@nxtensions/astro

### Describe the bug

Following this topic (https://github.com/nxtensions/nxtensions/issues/446), I wanted to integrate astro into my already created monorepo. When generating the astro application with Tailwind I saw that it was reporting this problem and tailwind was not working.

```
> nx run astro-blog:serve

  🚀  astro  v2.10.0 started in 301ms

  ┃ Local    http://localhost:3000/
  ┃ Network  use --host to expose


warn - The `content` option in your Tailwind CSS configuration is missing or empty.
warn - Configure your content sources or your generated CSS will be missing styles.
warn - https://tailwindcss.com/docs/content-configuration
```

### Expected behavior

I was expecting tailwind to be preconfigured and to work properly.

### Steps to reproduce

> $ pnpm i -D -E @nxtensions/astro
> $ pnpm exec nx generate @nxtensions/astro:application --name=astro-template --addCypressTests=false --integrations=tailwind --no-interactive
> $ nx dev astro-template

> astro.config.mjs
```
import { defineConfig } from 'astro/config'
import tailwind from '@astrojs/tailwind'

// https://astro.build/config
export default defineConfig({
  outDir: '../../dist/apps/astro-blog',
  integrations: [tailwind()],
})
```

> project.json
```
{
  "name": "astro-blog",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "application",
  "sourceRoot": "apps/astro-blog/src",
  "targets": {
    "build": {
      "outputs": ["{workspaceRoot}/dist/{projectRoot}"],
      "executor": "@nxtensions/astro:build",
      "options": {}
    },
    "serve": {
      "executor": "@nxtensions/astro:dev",
      "options": {}
    },
    "preview": {
      "dependsOn": [
        {
          "target": "build",
          "projects": "self"
        }
      ],
      "executor": "@nxtensions/astro:preview",
      "options": {}
    },
    "check": {
      "executor": "@nxtensions/astro:check"
    },
    "sync": {
      "executor": "@nxtensions/astro:sync"
    }
  },
  "tags": []
}

```

> tailwind.config.cjs
```
const { createGlobPatternsForDependencies } = require('@nxtensions/astro/tailwind')
const { join } = require('path')
const plugin = require('tailwindcss/plugin')

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    join(__dirname, 'src/**/!(*.stories|*.spec).{astro,html,js,jsx,md,svelte,ts,tsx,vue}'),
    ...createGlobPatternsForDependencies(__dirname),
  ],
  theme: {
    extend: {},
  },
  plugins: [
    plugin(function astroComponentsPlugin({ addComponents }) {
      addComponents({
        '.h-my-screen': {
          height: ['100vh', '100dvh'],
        },
      })
    }),
  ],
}
```

> tsconfig.json
```
{
  "extends": "astro/tsconfigs/base",
  "compilerOptions": {
    "strictNullChecks": true,
    "allowJs": true,
    "baseUrl": ".",
    "paths": {
      "@/*": ["src/*"]
    }
  }
}
```

### Environment

```
>  NX   Report complete - copy this into the issue template

   Node   : 18.16.0
   OS     : darwin-arm64
   pnpm   : 8.6.11

   nx (global)        : 16.5.5
   nx                 : 16.6.0
   @nx/js             : 16.6.0
   @nx/jest           : 16.6.0
   @nx/linter         : 16.6.0
   @nx/workspace      : 16.6.0
   @nx/cypress        : 16.6.0
   @nx/devkit         : 16.6.0
   @nx/eslint-plugin  : 16.6.0
   @nx/next           : 16.6.0
   @nx/react          : 16.6.0
   @nrwl/tao          : 16.6.0
   @nx/web            : 16.6.0
   typescript         : 5.1.6
   ---------------------------------------
   Community plugins:
   @nxtensions/astro : 16.0.0
   ---------------------------------------
   Local workspace plugins:
   	 @nx-monorepo/contentful
```

**Comment by jesusvallez** [comment]
@firxworx your astro.config.mjs works fine

```
import path from 'path'
import { workspaceRoot } from '@nx/devkit'
import { defineConfig } from 'astro/config'
import tailwind from '@astrojs/tailwind'

const APP_PATH = 'apps/astro-blog'
const APP_SRC_ROOT = path.resolve(workspaceRoot, APP_PATH, 'src')
const APP_DIST_ROOT = path.resolve(workspaceRoot, 'dist', APP_PATH)

// https://astro.build/config
export default defineConfig({
  outDir: APP_DIST_ROOT,
  srcDir: APP_SRC_ROOT,
  publicDir: path.resolve(APP_PATH, 'public'),

  integrations: [tailwind({ configFile: path.resolve(workspaceRoot, APP_PATH, 'tailwind.config.cjs') })],
})
```

![image](https://github.com/nxtensions/nxtensions/assets/19146593/5bb972b5-ac44-46da-99b7-4500e29db04b)

**Comment by leosvelperez** [comment]
Thanks for reporting this!

The Astro `tailwind` integration recently released a breaking change in its v4.0.0 which causes this issue (https://github.com/withastro/astro/blob/main/packages/integrations/tailwind/CHANGELOG.md#400). Quoting it "This changes the Tailwind config loading behaviour where it is loaded from process.cwd() instead of the project root".

I'll push a fix later to handle that and generate the right configuration. Existing projects will need to add the path to the configuration.
